akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -328,6 +347,36 @@ private[impl] final class AgentImpl[A <: Agent](
       .toVector
   }
 
+  private def mapSpiAgentException(func: Throwable => Any): Throwable => Any = {
+
+    def convert(thw: Throwable): Throwable = thw match {
+      case exc: SpiAgentException =>
+        try {
+          exc.reason match {
+            case ModelFailure              => new ModelException(exc.getMessage)
+            case RateLimitFailure          => new RateLimitException(exc.getMessage)
+            case TimeoutFailure            => new ModelTimeoutException(exc.getMessage)
+            case UnsupportedFeatureFailure => new UnsupportedFeatureException(exc.getMessage)
+            case InternalFailure           => new InternalServerException(exc.getMessage)
+
+            case OutputParsingFailure        => new OutputParsingException(exc.getMessage, exc.cause)
```

Comment:
In docs we have:
```
if (throwable instanceof JsonParsingException jsonParsingException) {
```

Either we unwrap and pass in the cause, or we make the slightly incompatible change and update the docs.

Reply:
We can just use JsonParsingException, I think:

* deserialize throws JsonParsingException as now
* runtime wraps it in AgentException using OutputParsingFailure
* we unwrap it back in the SDK and users get it as is

The different here is when we get a `OutputParsingFailure` we know that the cause will be `JsonParsingException`, to we return it as it.